signac/contrib/filterparse.py

```diff
@@ -6,30 +6,108 @@
 
 
 def _print_err(msg=None):
+    """Prints the error message.
+
+    Parameters
+    ----------
+    msg : str
+        Error message to be printed.
+         (Default value = None)
+
+    """
     print(msg, file=sys.stderr)
 
 
 def _with_message(query, file):
+    """Writes the message to the passed file.
+
+    Parameters
+    ----------
+    query : dict
+        Filter arguments.
+    file :
+        The file to write message for the query passed.
+
+    Returns
+    -------
+    query : dict
+        Filter arguments.
+
+    """
     print("Interpreted filter arguments as '{}'.".format(json.dumps(query)), file=file)
     return query
 
 
 def _read_index(project, fn_index=None):
+    """Read index from the file passed.
+
+    Parameters
+    ----------
+    project : class:`~.contrib.project.Project`
+        Signac project handle
```

Comment:
Small style note: we almost always avoid writing signac in uppercase (we say "The signac framework" or similar, to avoid confusion with the person Paul Signac). In this particular instance, it makes more sense to capitalize `Project` since that is the class.
```suggestion
        Project handle.
```